### PR TITLE
Merge rc0.1.1 branch back into master to pick up useful fixes

### PR DESCRIPTION
**Pull request by mrichar1**
There are a number of useful fixes in the rc0.1.1 branch, which was presumably started for the next release, but never fully completed.

I suggest merging these changes back into master for now, and then make a new release branch when we are ready to release.

This branch seems to mostly consists of test changes, but also includes a few updates to providers and small bug fixes.

### CODE REVIEW

**Comment by jensens** [comment]
What about keeping master the development head and branch away for each release as maintenance branches? IMO this is easier to maintain and it's more obvious where development takes place.

**Comment by mrichar1** [comment]
Yes - I agree about master/maintenance branches - it's the name of this branch that's confusing!

If we ignore the name of the branch, it's really just a number of useful commits that should get merged back into master (and completely ignore that it is named to look like a release).

**Comment by mrichar1** [comment]
OK - have done a review of the changes introduced in this PR - can't approve any commits though, so will summarise here.

1. Test changes: mostly simple config changes, no major new tests or features added. (at least one of which actually ran successfully on Travis CI!) I'd suggest merging these in without too much review, since they seem sensible and cause some previously failing tests to start working.

2. Provider changes: 
* TLS 1.2 support not available in py2.6, so code to handle this.
* oauth1: 
  * Yahoo: drop support for birthdate and gender 
* oauth2: 
  * DeviantArt: API url changes

3. Other changes:
*  Version bump to 0.1.1 in the docs/setup.py (5cb04a4) - might be worth removing these before merging?

**Comment by jensens** [comment]
version bump: the next release is probably a 0.2 or 1.0 (I would prefer semantic versioning and given BBB code was removed in a other PR: that is a major change)

**Comment by mrichar1** [comment]
Reverted the version bump since this merge isn't now anything to do with versions.

File: authomatic/providers/__init__.py [hunk 1]
```diff
@@ -392,8 +397,25 @@ def _fetch(self, url, method='GET', params=None, headers=None, body='', max_redi
...
+                    '{message}. The server {host} probably uses TLS 1.2 '
+                    'which is not supported by Python {version}.'
+                    .format(
+                        message=message,
+                        host=host,
+                        version=python_version,
+                    ))
+
+            original_message = e.message if hasattr(e, 'message') else str(e)
+            raise FetchError(message, original_message=original_message,
```
Comment (jensens) [inline authomatic/providers/__init__.py]: Python 2.6 is no longer supported.

File: authomatic/providers/oauth1.py [hunk 2]
```diff
@@ -967,6 +967,13 @@ class Vimeo(OAuth1):
     * Docs: https://developer.vimeo.com/apis/advanced#oauth-endpoints
     * API reference: https://developer.vimeo.com/apis
 
+    .. note::
+
+        Vimeo seems to be using TLS 1.2 for SSL connection, which is not
+        supported by Python 2.6. For more info read `this stackoverflow question
+        <http://stackoverflow.com/questions/32853778/
+        python-sslv3-alert-handshake-failure-when-scraping-a-site>`__.
+
```
Comment (jensens) [inline authomatic/providers/oauth1.py]: same here, no Python 2.6 support.

File: authomatic/providers/oauth2.py [hunk 3]
```diff
@@ -356,13 +356,15 @@ def login(self):
...
-                                                             headers=self.access_token_headers)
+            request_elements = self.create_request_elements(
+                request_type=self.ACCESS_TOKEN_REQUEST_TYPE,
+                credentials=self.credentials,
+                url=self.access_token_url,
+                method=self.token_request_method,
+                redirect_uri=self.url,
+                params=self.access_token_params,
+                headers=self.access_token_headers
+            )
```
Comment (jensens) [inline authomatic/providers/oauth2.py]: this formatting changes are already in the cleanup branch.

File: authomatic/providers/oauth2.py [hunk 4]
```diff
@@ -371,11 +373,20 @@ def login(self):
...
+
+                # TODO: Check whether provider requires the User-Agent header
+                # if so, use more meaningful message.
+
+                raise FailureError(
+                    message=message,
+                    original_message=response.content,
+                    status=response.status,
+                    url=self.access_token_url,
+                )
```
Comment (jensens) [inline authomatic/providers/oauth2.py]: formatting changes already in the cleanup branch.

File: tests/functional_tests/config-template.py [hunk 5]
```diff
@@ -83,29 +85,38 @@ def teardown():
...
     'openid_yahoo',
 ]
 
+# Exclude providers who use TLS 1.2 from Python 2.6 tests
+if sys.version.startswith('2.6'):
+    EXCLUDED = EXCLUDED + [
+        'deviantart',
+        'paypal',
+        'reddit',
+    ]
```
Comment (jensens) [inline tests/functional_tests/config-template.py]: I think we should just drop 2.6 support.

File: tests/functional_tests/config-template.py [hunk 6]
```diff
@@ -83,29 +85,38 @@ def teardown():
...
+
+# There are some problems with OpenSSL on Python 3 on OSX
+if sys.version.startswith('3') and sys.platform == 'darwin':
+    EXCLUDED = EXCLUDED + [
+        'paypal',
+    ]
+
```
Comment (jensens) [inline tests/functional_tests/config-template.py]: Is this still an issue? I know many people using Python 3 on MacOS with SSL.

**Comment by mrichar1** [comment]
Have cherry-picked the 'provider bug' commits into new PRs:

DeviantArt 699b453 #179
Reddit b0109af #180
Yahoo 86852d7 #181

**Comment by jensens** [comment]
Great! All merged, I think we are done with this one, I close it.
@mrichar1 if you also think all relevant commits are now merged, please delete the branch.

**Comment by mrichar1** [comment]
Have cherry-picked the 'test fixes' into a new PR: #183 - will now delete the rc0.1.1 branch as there's nothing else useful left in it.

**Pull request closed without merging**